hypothesis-python/scripts/basic-test.sh

```diff
@@ -35,7 +27,10 @@ else
   $PYTEST "tests/py3"
 fi
 
-if [ "$DARWIN" = true ]; then
+# We run a reduced set of tests on the macOS CI so that it runs in vaguely
+# reasonable time.
+if [ "$CIRCLECI" = true ]; then
```

Comment:
Whoops, I'd somehow missed that variable. Definitely better than the one I suggested.

It's just occurred to me that this is probably not going to work (sorry) because this is run from inside tox, which discards environment variables by default. You need to add `CIRCLECI` to https://github.com/HypothesisWorks/hypothesis/blob/master/hypothesis-python/tox.ini#L10

Sorry about this! CI involves a lot of moving parts, and I always manage to forget just how many.